app/scripts/controllers/network/network-controller.js

```diff
@@ -125,7 +125,9 @@ export default class NetworkController extends EventEmitter {
     }
     this._infuraProjectId = infuraProjectId;
 
-    this.on(NETWORK_EVENTS.NETWORK_DID_CHANGE, this.lookupNetwork);
```

Comment:
Huh. I'm guessing this would fail each time because it was called unbound, and `this` wasn't set to anything.

I looked back in the git history and this was _always_ called this way, so.... was this always broken? :thinking:

Reply:
Perhaps better off removing it if it never worked. Especially for this PR.

Reply:
Ah. I had fixed this method call early on, then when I went to revert this change, it broke a bunch of tests. 

If this is broken, it means that when the network is switched via `setRpcTarget` or `setNetworkType`, rolled back via `rollbackToPreviousProvider`, or reset via `resetConnection`, a new provider gets created, but the network details, which holds whether the network supports EIP-1559, don't get reset until the next time that `getEIP1559Compatibility` is called. So this doesn't seem like the best design to me. 

I suppose considering that's how it works right now, though, I will remove this. On the `core` side, though, it looks like `lookupNetwork` is called directly in `setProviderType` and `setRpcTarget` as opposed to being indirectly called through an event, so I'm wondering if that's how this controller should work.

Reply:
@Gudahtt So it turns out that this always _did_ work, it's just not obvious. It looks like the EventEmitter module will implicitly call the function it's given with the same instance that the event listener is created on: https://nodejs.org/docs/latest-v16.x/api/events.html#passing-arguments-and-this-to-listeners. So `this` is set to the NetworkController in this case.

Changing this code to use an explicit arrow function happened to make that was previously failing without this change pass, but it turns out that that test was incorrect. I've fixed this test and reverted the change to this line: [02daaa7](https://github.com/MetaMask/metamask-extension/pull/17498/commits/02daaa715c14f9cdb9fa7bb45424e0851e77235a)

Reply:
Nice! Interesting find, I had no idea it worked that way. That would makes sense though, I would have been surprised if this bug had been here for so long.